src/BloomExe/web/ServerBase.cs

```diff
@@ -116,117 +73,119 @@ private bool StartWithExceptionHandling(out Exception error)
 			catch (Exception e)
 			{
 				error = e;
-				_listener = null;
 				return false;
 			}
 		}
 
 		private bool TryStart()
 		{
-			_listener = new HttpListener();
 			_listener.AuthenticationSchemes = AuthenticationSchemes.Anonymous;
 			_listener.Prefixes.Add(PathEndingInSlash);
 			_listener.Start();
-			//			_listener.BeginGetContext(new AsyncCallback(ListenerCallback), _listener);
 
-			_listenerThread = new Thread(HandleRequests);
 			_listenerThread.Start();
 
-			return GetIsAbleToUsePort();
-		}
-
-		private bool GetIsAbleToUsePort()
-		{
-			IsAbleToUsePort = false;
-			try
+			for (var i = 0; i < _workers.Length; i++)
 			{
-				var x = new WebClientWithTimeout { Timeout = 3000 };
-
-				IsAbleToUsePort = ("OK" == x.DownloadString(PathEndingInSlash + "testconnection"));
+				_workers[i] = new Thread(Worker);
+				_workers[i].Start();
 			}
-			catch (Exception e)
-			{
-				IsAbleToUsePort = false;
-			}
-			return IsAbleToUsePort;
+
+			return GetIsAbleToUsePort();
 		}
 
 		private void HandleRequests()
 		{
 			while (_listener.IsListening)
 			{
-				var context = _listener.BeginGetContext(new AsyncCallback(ListenerCallback), _listener);
+				var context = _listener.BeginGetContext(ContextReady, null);
+
 				if (0 == WaitHandle.WaitAny(new[] { _stop, context.AsyncWaitHandle }))
 					return;
 			}
 		}
 
-		/// <summary>
-		/// TODO: Note: doing this at runtime isn't as good as doing it in the installer, because we have no way of
-		/// removing these entries on uninstall (but the installer does).
-		/// </summary>
-		private static void AddUrlAccessControlEntry()
+		private void ContextReady(IAsyncResult ar)
 		{
-			MessageBox.Show(
-				string.Format("We need to do one more thing before Bloom is ready. Bloom needs temporary administrator privileges to set up part of its communication with the embedded web browser.{0}{0}After you click 'OK', you may be asked to authorize this step.", Environment.NewLine),
-				"Almost there!", MessageBoxButtons.OK);
-
-			var startInfo = new System.Diagnostics.ProcessStartInfo();
-			startInfo.UseShellExecute = true;
-			startInfo.Verb = "runas"; //makes it as for elevation to admin rights
-
-			if (Environment.OSVersion.Version.Major == 5 /*win xp*/)
-			{
-				startInfo.FileName = "httpcfg";
-				startInfo.Arguments = string.Format("set urlacl -u {0} /a \"D:(A;;GX;;;WD)\"", PathEndingInSlash);
-			}
-			else
+			try
 			{
-				startInfo.FileName = "netsh";
-				startInfo.Arguments = string.Format("http add urlacl url={0} user=everyone", PathEndingInSlash.TrimEnd('/'));
+				lock (_queue)
+				{
+					_queue.Enqueue(_listener.EndGetContext(ar));
+					_ready.Set();
+				}
 			}
-			System.Diagnostics.Process.Start(startInfo);
+			catch { }
```

Comment:
Why do you need to catch ALL exceptions here? Any chance of making it more specific? Surely there are some things that could go wrong here that we might want to know about?